diff -u -r origin/lib/html.php after/lib/html.php
--- origin/lib/html.php	Tue May 10 21:55:39 2022
+++ after/lib/html.php	Thu May 12 01:19:47 2022
@@ -227,6 +227,7 @@
 	global $external_link_cushion_page, $external_link_cushion;
 	global $topicpath_title;
 	global $ticket_jira_default_site;
+	global $autohashtaglink;
 	global $show_passage;
 	if (!isset($ticket_link_sites) || !is_array($ticket_link_sites)) {
 		return '';
@@ -296,6 +297,11 @@
 <input type="hidden" class="ticketlink-jira-default-def" value="$h_ticket_jira_default_site" />
 EOS;
 	}
+	// AutoHashTagLink
+	$autohashtaglink_value = ($plugin == "read") ? $autohashtaglink : 0;
+	$autohashtaglink_data = <<<EOS
+<input type="hidden" class="auto-hashtag-link" value="$autohashtaglink_value" />
+EOS;
 	// External link cushion page
 	$external_link_cushion_data = '';
 	if ($external_link_cushion_page) {
@@ -323,6 +329,7 @@
 $ticketlink_data
 $ticketlink_jira_data
 $ticketlink_jira_default_data
+$autohashtaglink_data
 $external_link_cushion_data
 $topicpath_data
 </div>
diff -u -r origin/pukiwiki.ini.php after/pukiwiki.ini.php
--- origin/pukiwiki.ini.php	Tue May 10 21:55:39 2022
+++ after/pukiwiki.ini.php	Thu May 12 01:27:18 2022
@@ -177,6 +177,11 @@
 $autoalias_max_words = 50; // pairs
 
 /////////////////////////////////////////////////
+// AutoHashTagLink feature
+// Automatic link to hash tag
+$autohashtaglink = 2; // Bytes, 0 = OFF (try 2)
+
+/////////////////////////////////////////////////
 // Enable Freeze / Unfreeze feature
 $function_freeze = 1;
 
diff -u -r origin/skin/main.js after/skin/main.js
--- origin/skin/main.js	Tue May 10 21:55:39 2022
+++ after/skin/main.js	Thu May 12 02:25:09 2022
@@ -323,6 +323,21 @@
     var target = document.getElementById('body')
     walkElement(target)
   }
+  function autoHashTagLink() {
+    var autoHashTagLink = document.querySelector('#pukiwiki-site-properties .auto-hashtag-link')
+    if (autoHashTagLink.value === "0") return
+    var propsE = document.querySelector('#pukiwiki-site-properties .site-props')
+    if (!propsE || !propsE.value) return
+    var siteProps = JSON.parse(propsE.value)
+    var sitePathname = siteProps && siteProps.base_uri_pathname
+    if (!sitePathname) return
+    var element = document.querySelector("#contents")
+    var re = new RegExp('\#([A-Za-z0-9]{' + autoHashTagLink.value + ',})', 'g')
+    var re2 = new RegExp('(?<=>)[^<>]*?(' + '\#([A-Za-z0-9]{' + autoHashTagLink.value + ',})' + ')[^<>]*?(?=<)', 'g')
+    element.innerHTML = element.innerHTML.replace(re2, function() {
+      return arguments[0].replace(re, "<a href=\"" + sitePathname + "?$1\">#$1</a>")
+    })
+  }
   function confirmEditFormLeaving () {
     function trim (s) {
       if (typeof s !== 'string') {
@@ -667,6 +682,7 @@
   }
   setYourName()
   autoTicketLink()
+  autoHashTagLink()
   confirmEditFormLeaving()
   showPagePassage()
   convertExternalLinkToCushionPageLink()
